go/vt/vtgate/vcursor_impl.go

```diff
@@ -1200,3 +1207,39 @@ func (vc *vcursorImpl) StorePrepareData(stmtName string, prepareData *vtgatepb.P
 func (vc *vcursorImpl) GetPrepareData(stmtName string) *vtgatepb.PrepareData {
 	return vc.safeSession.GetPrepareData(stmtName)
 }
+
+func (vc *vcursorImpl) GetWarmingReadsPercent() int {
+	return vc.warmingReadsPercent
+}
+
+func (vc *vcursorImpl) CloneForReplicaWarming(ctx context.Context) interface{} {
+	callerId := callerid.EffectiveCallerIDFromContext(ctx)
+	immediateCallerId := callerid.ImmediateCallerIDFromContext(ctx)
+
+	timedCtx, _ := context.WithTimeout(context.Background(), 5*time.Second)
+	clonedCtx := callerid.NewContext(timedCtx, callerId, immediateCallerId)
+
+	v := &vcursorImpl{
+		safeSession:         NewAutocommitSession(vc.safeSession.Session),
+		keyspace:            vc.keyspace,
+		tabletType:          topodatapb.TabletType_REPLICA,
+		destination:         vc.destination,
+		marginComments:      vc.marginComments,
+		executor:            vc.executor,
+		resolver:            vc.resolver,
+		topoServer:          vc.topoServer,
+		logStats:            &logstats.LogStats{Ctx: clonedCtx},
+		collation:           vc.collation,
+		ignoreMaxMemoryRows: vc.ignoreMaxMemoryRows,
+		vschema:             vc.vschema,
+		vm:                  vc.vm,
+		semTable:            vc.semTable,
```

Comment:
I think I spot a handful of `SemTable` functions that modify state. This may be completely fine, I guess it depends how and when `SemTable` is used, or may be moot because as far as I can tell nothing actually calls `GetSemTable`.

But to be totally safe, what you could try:
 * Remove `GetSemTable` from the interface entirely.
 * Panic when `GetSemTable` is called.
 * Use a cloned `SemTable` here.